lib/zlib.js

```diff
@@ -251,9 +251,20 @@ function Zlib(opts, mode) {
       opts.finishFlush, 'options.finishFlush',
       Z_NO_FLUSH, Z_BLOCK, Z_FINISH);
 
-    windowBits = checkRangesOrGetDefault(
-      opts.windowBits, 'options.windowBits',
-      Z_MIN_WINDOWBITS, Z_MAX_WINDOWBITS, Z_DEFAULT_WINDOWBITS);
+      // windowBits is special. On the compression side, 0 is an invalid value.
+      // But on the decompression side, a value of 0 for windowBits tells zlib
+      // to extract the value from the headers of the compressed stream.
+    if (opts.windowBits === 0 &&
```

Comment:
We may actually want to also capture `undefined` here, so that when decompressing we always go with the header-provided value (instead of the safe maximum) for lower memory usage.

Reply:
excellent idea! How about `opts.windowBits == null || opts.windowBits === 0`?

Reply:
@anandsuresh Sounds reasonable to me!